### DISCUSSION TITLE
Why is hardware acceleration disabled in linux?

### DISCUSSION THREAD

**Discussion by apalmerm** [cypress-io/cypress#28883] [Questions and Help]
While looking through the electron code to get an idea of how electron is being launched behind the scenes we noticed that in linux, hardware acceleration is being disabled:
<img width="797" alt="hardwareAccelerationDisabled" src="https://github.com/cypress-io/cypress/assets/158522116/6f34ab35-06fa-447a-b451-b0bb45b6d223"> (from this file: https://github.com/cypress-io/cypress/blob/develop/packages/server/lib/environment.js)

It looks like chrome disables the hardware acceleration as well: 
![chromeHardwareAccelerationDisabled](https://github.com/cypress-io/cypress/assets/158522116/4c7da32e-8b4c-4add-a5db-e3a4dff909e6)

I was wondering a few things about this:
1. Am I understanding this code correctly that hardware acceleration is being disabled for linux (as in that option is working as I would expect it to and isn't being overwritten anywhere)?
2. Why is hardware acceleration being disabled for linux?
3. Is disabling the hardware acceleration a necessity? In other words, what (if anything) would happen if that line was removed (so that we were no longer disabling the hardware acceleration)?

Thank you in advance for any insights on this!

**Comment by portyu9** [comment]
Hi @apalmerm 

Yes — Cypress currently disables hardware acceleration for its **Electron browser on Linux**.

The important distinction is that this is an Electron API:

```js
app.disableHardwareAcceleration()
```

so it applies to Cypress's Electron process. It does **not** mean Cypress automatically disables GPU acceleration when you explicitly run a separate Chrome/Chromium browser.

The current Cypress source still contains the Linux-specific call, so this behavior was not just an old Cypress 13 implementation detail.

As for **why**, I don't see the exact rationale documented alongside the code, so I would not claim a specific root cause as fact. Stability across Linux/headless/CI GPU and driver environments is a plausible reason, but that is an inference rather than something established by this thread.

Removing the line would require running a modified Cypress build. That is not a supported configuration and could expose rendering/GPU/driver issues that the standard Cypress Electron setup intentionally avoids.

If your application specifically needs hardware-accelerated behavior such as WebGL/GPU rendering, I would test it using Chrome/Chromium rather than patching Cypress's Electron startup code.

Please mark this answer as accepted if it helped, thank you.
